File: 04-mongo-intro/mongo-client.py
from pymongo import MongoClient

mongo_uri = "mongodb://localhost:27017"

client = MongoClient(mongo_uri)
db = client['mptibd']

print("Collections:", db.list_collection_names())

alunos = db.get_collection('alunos').find()
for aluno in alunos:
    print(aluno)
    print("Nome:", aluno['nome'])
    print("Idade:", aluno['idade'])
    print("Curso:", aluno['curso'])
    print("Notas:", aluno['notas'])
    print()

db.get_collection('alunos').insert_one({
    'nome': 'José da Silva Xavier',
    'idade': 50,
    'curso': 'Engenharia de Software',
    'notas': [8.5, 9.0, 7.5]
})

db.get_collection('alunos').delete_many({
    'idade': { '$gte': 50 }
})

db.get_collection('xyz').insert_one({
    'nome': 'José da Silva Xavier',
    'idade': 50,
    'curso': 'Engenharia de Software',
    'notas': [8.5, 9.0, 7.5]
})

resultado = db.get_collection('alunos').aggregate([
    { '$match': { 'curso': 'Engenharia de Software' } },
    { '$project': { 'nome': 1, 'notas': 1 } },
    {
        '$group': {
            '_id': '$curso',
            'media': { '$avg': '$notas' }
        }
    },
    {
        '$sort': { 'media': -1 }
    }
])

for doc in resultado:
    print(doc)

File: 04-mongo-intro/mongo-client.js
db.getCollection("alunos").insertMany([
    { nome: "Lucas", idade: 20, curso: "Engenharia de Software" },
    { nome: "Ana", idade: 22, curso: "Ciência da Computação" },
    { nome: "Pedro", idade: 19, curso: "Sistemas de Informação" },
    { nome: "Maria", idade: 21, curso: "Engenharia de Dados" },
    { nome: "João", idade: 23, curso: "Engenharia de Software" },
    { nome: "Fernanda", idade: 20, curso: "Ciência da Computação" },
    { nome: "Carlos", idade: 22, curso: "Sistemas de Informação" },
    { nome: "Juliana", idade: 19, curso: "Engenharia de Dados" }
]);

use("mptibd");
db.getCollection("alunos").find({}).pretty();

use("mptibd");
db.getCollection("alunos").insertOne({
    nome: "João Victor",
    idade: 20,
    curso: "Mestrado em Tecnologia da Informação",
    notas: [
        { disciplina: "Matemática", nota: 8.5 },
        { disciplina: "Programação", nota: 9.0 },
        { disciplina: "Banco de Dados", nota: 7.5 }
    ]
})

db.getCollection("alunos").find({
    idade: { $lt: 21 },
}).pretty();

use("mptibd");
db.getCollection("alunos").updateMany(
    { notas: { $exists: false } },
    {
        "$set": { notas: [
        {
            disciplina: "Matemática",
            nota: 0
        },
        {
            disciplina: "Programação",
            nota: 0
        },
        {
            disciplina: "Banco de Dados",
            nota: 0
        }]
    }
})

use("mptibd");
db.getCollection("alunos").updateMany(
    { notas: { $elemMatch: { nota: 0 } } },
    {
        "$set": { notas: [
        {
            disciplina: "Matemática",
            nota: 10
        },
        {
            disciplina: "Programação",
            nota: 10
        },
        {
            disciplina: "Banco de Dados",
            nota: 10
        }]
    }
})

use("mptibd");
db.getCollection("alunos").deleteOne(
    { "nome": { "$eq": "Carlos" }}
)


use("mptibd");
db.getCollection("alunos").deleteMany(
    { "idade": { "$lt": 20 }}
)
